src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -359,6 +367,73 @@ def get_package_size(self, output):
             return Constants.UNKNOWN_PACKAGE_SIZE
     # endregion
 
+    # region auto OS updates
+    def disable_auto_os_update(self):
+        """ Disables auto OS updates on the machine only if they are enabled and logs the default settings the machine comes with """
+        try:
+            self.composite_logger.log_debug("Disabling auto OS updates if they are enabled")
+            self.backup_image_default_patch_mode()
+            self.update_image_default_patch_mode(self.update_package_list, "0")
+            self.update_image_default_patch_mode(self.unattended_upgrade, "0")
+            self.composite_logger.log("Successfully disabled auto OS updates")
+        except Exception as error:
+            self.composite_logger.log_error("Could not disable auto OS updates. [Error={0}]".format(repr(error)))
+
+    def backup_image_default_patch_mode(self):
+        """ Records the default system settings for auto OS updates within patch extension artifacts for future reference.
+        We only log the default patch_mode a VM comes with, any subsequent updates will not be recorded"""
+        try:
+            if not self.image_default_patch_mode_backup_exists():
+                image_default_patch_mode = self.env_layer.file_system.read_with_retry(self.image_default_patch_mode_file_path)
+                settings = image_default_patch_mode.strip().split('\n')
+                for patch_mode in settings:
+                    if self.update_package_list in str(patch_mode):
+                        self.update_package_list_value = re.search(self.update_package_list + ' *"(.*?)".', str(patch_mode)).group(1)
+                    if self.unattended_upgrade in str(patch_mode):
+                        self.unattended_upgrade_value = re.search(self.unattended_upgrade + ' *"(.*?)".', str(patch_mode)).group(1)
```

Comment:
Needs more logging. For any significant line that can fail, logging must exist prior to its execution. The outcome must be recorded after completion. For any code path that will execute only in an if condition, the opposite condition must be logged in an else block to explain execution flow only looking at logs.

This comment generally applies to all code - review for applicability everywhere.